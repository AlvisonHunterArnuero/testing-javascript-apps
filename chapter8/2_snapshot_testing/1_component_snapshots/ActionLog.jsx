import React from "react";

export const ActionLog = ({ actions }) => {
  return (
    <div data-testid="action-log">
      <h2>Action Log</h2>
      <ul>
        {actions.map(({ time, message, data }, i) => {
          const date = new Date(time).toUTCString();
          return (
            <li key={i}>
              Date: {date} - Message: {message} - Data: {JSON.stringify(data)}
            </li>
          );
        })}
      </ul>
    </div>
  );
};
