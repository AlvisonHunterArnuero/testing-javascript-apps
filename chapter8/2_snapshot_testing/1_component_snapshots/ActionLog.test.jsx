import React from "react";
import { ActionLog } from "./ActionLog";
import { render } from "@testing-library/react";

const daysToMs = days => days * 24 * 60 * 60 * 1000;

test("logging actions", () => {
  const actions = [
    {
      time: new Date(daysToMs(1)),
      message: "Loaded item list",
      data: { cheesecake: 2, macaroon: 5 }
    },
    {
      time: new Date(daysToMs(2)),
      message: "Item added",
      data: { cheesecake: 2 }
    },
    {
      time: new Date(daysToMs(3)),
      message: "Item removed",
      data: { cheesecake: 1 }
    },
    {
      time: new Date(daysToMs(4)),
      message: "Something weird happened",
      data: { error: "The cheesecake is a lie" }
    }
  ];

  const { container } = render(<ActionLog actions={actions} />);
  expect(container).toMatchSnapshot();
});
