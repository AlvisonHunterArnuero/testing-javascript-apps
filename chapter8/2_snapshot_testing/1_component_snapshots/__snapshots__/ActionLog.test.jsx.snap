// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logging actions 1`] = `
<div>
  <div
    data-testid="action-log"
  >
    <h2>
      Action Log
    </h2>
    <ul>
      <li>
        Date: 
        Fri, 02 Jan 1970 00:00:00 GMT
         - Message: 
        Loaded item list
         - Data: 
        {"cheesecake":2,"macaroon":5}
      </li>
      <li>
        Date: 
        Sat, 03 Jan 1970 00:00:00 GMT
         - Message: 
        Item added
         - Data: 
        {"cheesecake":2}
      </li>
      <li>
        Date: 
        Sun, 04 Jan 1970 00:00:00 GMT
         - Message: 
        Item removed
         - Data: 
        {"cheesecake":1}
      </li>
      <li>
        Date: 
        Mon, 05 Jan 1970 00:00:00 GMT
         - Message: 
        Something weird happened
         - Data: 
        {"error":"The cheesecake is a lie"}
      </li>
    </ul>
  </div>
</div>
`;
